### ISSUE TITLE
Bazel: Updating http_file's that point into GitHub repos

### ISSUE THREAD

**Issue by cpsauer** [renovatebot/renovate#8784]
**What would you like Renovate to be able to do?**

I'd love it if Renovate could update Bazel http_file rules that point to raw.githubusercontent.

There's a handy use case where you grab a build rule (or other file) from another repo and use it in yours, and it's be great to have Renovate watch for changes. Here's a WORKSPACE example:
```
http_file(
    name = "eigen.BUILD",
    urls = ["https://raw.githubusercontent.com/tensorflow/tensorflow/8ab9ab6217aba0b771f05b3bd883ee33ae860931/third_party/eigen.BUILD"],
    sha256 = "a4ca1ae2560b6e371cc862ab9b43c7828f65827096efe0b0c11fc819a12fd9a5",
)
```

**Did you already have any implementation ideas?**

I think the simplest would be to download the latest version of the file, compare it to the hash of the existing, and update if they differ. 

[The latest lives at `https://raw.githubusercontent.com/tensorflow/tensorflow/master/third_party/eigen.BUILD` in this case, but it could just be a matter of getting the main branch's name and swapping that in for the hash.]

**Other notes**

This is definitely a more niche use case, but a handy one. Definitely lower priority than https://github.com/renovatebot/renovate/issues/7945 but wanted to file anyway.

Thanks for your consideration, and for a great product!

**Comment by rarkins** [comment]
The way we'd normally fit this into our workflow is:
- Great the dependency as `github-refs` with depName=tensorflow/tensorflow and currentValue=8ab9ab6217aba0b771f05b3bd883ee33ae860931
- Check the commit of master in tensorflow/tensorflow and update the WORKSPACE if it's changed
- Update the corresponding sha

The problem with that is that you don't want to be changing the commit hash unnecessarily if the file contents haven't changed. A proxy for the file contents is the sha256, but we don't currently have any awareness of "don't change the version if the sha is unchanged" so would need to work out how that fits in.

**Comment by cpsauer** [comment]
👍🏻 With you. 

Did a quick check that the raw file can be accessed on commits that don't change it (and it can).

Seems like the check for file content has unchanged could be added here.
https://github.com/renovatebot/renovate/blob/05ecddf8052d332f210437d5df663ca43a8093a4/lib/manager/bazel/update.js#L59-L64
Just returning null and logging info like in this case:
https://github.com/renovatebot/renovate/blob/05ecddf8052d332f210437d5df663ca43a8093a4/lib/manager/bazel/update.js#L81-L84

**Comment by rarkins** [comment]
The way Renovate's current logic works, we could "return null" from an update to abort it if we conclude that the update shouldn't be made. It would result in a "no files to commit" type of log message. There are some undesirable side effects though:
- The update would be re-attempted every single run
- The branch and update would show up in logs and Dependency Dashboard as pending, e.g. people could mistakenly think they're out of date
- If the update were grouped with other updates which *didn't* abort then the PR body would list the aborted update

We could maybe improve this with a moderate tweak to existing logic:
- Create a new datasource called `github-file` which takes the full repo and file path as `depName`
- The bazel extract saves the commit hash as `currentValue` while the sha256 is `currentDigest` (fairly normal behavior)
- The datasource returns the latest commit has for the repo as `newValue` and the file's sha256 as `newDigest` (this is normal for a datasource)
- We have new logic in our lookup logic which says "don't update a current value if the digest is unchanged", so that no update is recorded as pending unless the digest changes

Another potential approach would be to treat the sha256 as the currentValue and the commit hash as digest, which could potentially even fit in with the current value/digest logic.

**Comment by rarkins** [comment]
Blocked by #9667
